src/Web/Form.hs

```diff
@@ -0,0 +1,31 @@
+module Web.Form (
+  module Web.Form,
+  module Web.Form.Types,
+) where
+
+import Data.Map hiding (fold, toList)
+import ImportYesod
+import Templates.Partials.FlashAlerts
+import Web.FlashAlerts
+import Web.Form.Types
+
+fieldError :: e -> FormValidation e a
+fieldError err = FieldErrors $ err :| []
+
+lookupFormFieldTextError :: Text -> Map Text Text -> FormValidation Text Text
+lookupFormFieldTextError k m = lookupFormField id k m
+
+lookupFormField :: (Text -> e) -> Text -> Map Text Text -> FormValidation e Text
+lookupFormField err k m =
+  case lookup k m of
+    Nothing -> fieldError . err $ k
+    Just v  -> Result v
+
+lookupOptionalFormField :: Text -> Map Text Text -> FormValidation Text (Maybe Text)
+lookupOptionalFormField k m = Result $ lookup k m
```

Comment:
A lookup function that allows you to continue chaining within the `FormValidation` applicative / monad, but that doesn't trip up the form validation when the field is not present.